pkg/s3select/json/reader.go

```diff
@@ -105,20 +106,45 @@ func (or *objectReader) objectEndIndex(p []byte, length int) int {
 	return -1
 }
 
+// Updates internal states depending on cached values etc.
+// This function updates start and end offsets of JSON objects
+// which is kept such that we can always properly return
+// right JSON objects.
+func (or *objectReader) updateState(p []byte, n int) int {
+	index := or.findEndIndex(p[:n])
```

Comment:
You could pass `p` and `n` to avoid trimming.

Reply:
In this scenario trimming is fine, as we are not interested in the entire p. You only read upto `n` as per implementation.